### PR TITLE
Fixed simplifyerror in trigsimp

### LINKED ISSUE #17633: Sympy simplify() does not act just by having a variable multiplying some trigonometric expressions
Python3 version is 3.7.3-1
Sympy version is 1.3-2
Scipy version is 1.2.2-7

I show an expression which is zero after simplifying. Then I just add a "-1" and the new expression is not simplified to "-1". I also observe that it is simplified to "-1" if "b" is either removed or substituted by any number. And this should not be necessary, it should work in any case.

So there is something about multiplications and additions/substractions that confuses sympy.simplify().

Here is the "ejemplo.py" code:

```
import sympy as sp

a,b = sp.symbols( 'a,b', real = True )

x = 4 * b * sp.sin(a) ** 3 - 3 * b * sp.sin(a) + b * sp.sin(3*a)

print('Before sp.simplify           :', x   )
y = sp.simplify(x)
print('After sp.simplify            :', y   )

print('')

print('Expression -1                :', x-1 )
z = sp.simplify(x - 1)
print('Effect of -1 over sp.simplify:', z   )

print('\nIf b is removed, -1 is obtained. But it should not be necessary')
x1 = 4 * sp.sin(a) ** 3 - 3 * sp.sin(a) + sp.sin(3*a) -1
x1 = sp.simplify(x1)
print('Expected output (removing)   :', x1 )

x2 = (x-1).subs({b:823910})
x2 = sp.simplify(x2)
print('Expected output (any subs)   :', x2 )
```

Code output:

```
Before sp.simplify           : 4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a)
After sp.simplify            : 0

Expression -1                : 4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a) - 1
Effect of -1 over sp.simplify: 4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a) - 1

If b is removed, -1 is obtained. But it should not be necessary
Expected output (removing)   : -1
Expected output (any subs)   : -1


------------------
(program exited with code: 0)
Press return to continue
```

### PR DESCRIPTION
**Pull request by Smit-create**
Fixed simplify error in trigsimp to evaluate result correctly
#### References to other Issues or PRs

Fixes #17633

#### Brief description of what is fixed or changed
Before:
```
>>> a,b,c = symbols( 'a,b,c', real = True)
>>> x = 4 * b * sin(a) ** 3 - 3 * b * sin(a) + b * sin(3*a)
>>> print(trigsimp(x - c))
4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a) - c
```

After:
```
>>> a,b,c = symbols( 'a,b,c', real = True)
>>> x = 4 * b * sin(a) ** 3 - 3 * b * sin(a) + b * sin(3*a)
>>> print(trigsimp(x - c))
- c
```

#### Release Notes

* simplify
  * Fixes to prevent loss of result produced by TR8

### CODE REVIEW

File: sympy/simplify/tests/test_trigsimp.py [hunk 1]
```diff
@@ -218,6 +218,10 @@ def test_trigsimp_issue_4032():
 def test_trigsimp_issue_7761():
     assert trigsimp(cosh(pi/4)) == cosh(pi/4)
 
+def test_trigsimp_issue_17633():
+    a,b,c = symbols( 'a,b,c', real = True )
```
Comment (jksuom) [inline sympy/simplify/tests/test_trigsimp.py]: Do these have to be real?

Comment (Smit-create) [reply sympy/simplify/tests/test_trigsimp.py]: No, I will change that. I works fine without that.

File: sympy/simplify/tests/test_trigsimp.py [hunk 2]
```diff
@@ -218,6 +218,10 @@ def test_trigsimp_issue_4032():
...
+    x = 4 * b * sin(a) ** 3 - 3 * b * sin(a) + b * sin(3*a)
```
Comment (jksuom) [inline sympy/simplify/tests/test_trigsimp.py]:
```suggestion
    x = 4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a)
```

File: sympy/simplify/tests/test_trigsimp.py [hunk 3]
```diff
@@ -219,8 +219,8 @@ def test_trigsimp_issue_7761():
     assert trigsimp(cosh(pi/4)) == cosh(pi/4)
 
 def test_trigsimp_issue_17633():
-    a,b,c = symbols( 'a,b,c', real = True )
-    x = 4 * b * sin(a) ** 3 - 3 * b * sin(a) + b * sin(3*a)
+    a,b,c = symbols( 'a,b,c')
```
Comment (jksuom) [inline sympy/simplify/tests/test_trigsimp.py]:
```suggestion
    a, b, c = symbols('a,b,c')
```

Comment (Smit-create) [reply sympy/simplify/tests/test_trigsimp.py]: Sorry for that, I corrected it.

**Comment by jksuom** [comment]
It looks like there should be an `expand()` method for Tuples. It could probably be defined as a no-op fall-back for all Basic objects:

    def expand(self, **kwargs):
        return self

**Comment by oscarbenjamin** [comment]
I'd prefer `Tuple.expand` to expand the elements of the `Tuple` (#18181)

**Comment by jksuom** [comment]
> I'd prefer Tuple.expand to expand the elements of the Tuple

The elements of a Tuple need not have an expand method (e.g. an ExprCondPair may contain True).  It would be possible to expand the elements if they are checked.  But I'm not sure what would be the best way to do that. That is why I suggested the no-op fall-back.

**Comment by oscarbenjamin** [comment]
That seems reasonable as a default for `Basic` (assuming we want everything to have an `expand` method).

As a broader fix perhaps it would be good if trigsimp didn't try to apply its algorithms directly to non-Expr types.

**Comment by czgdp1807** [comment]
Any updates on this?

**Comment by czgdp1807** [comment]
@Smit-create Are you working on it?

**Comment by Smit-create** [comment]
I am closing this, for now. If anyone wants to work on this feel free to take over.

**Pull request closed without merging**